.azure-pipelines/pull_request.yml

```diff
@@ -10,97 +10,201 @@ pr:
+              ArtifactName: 'documentation'
+            displayName: 'Upload built documentation'
 
       - job: 'mac'
         displayName: 'Mac OS'
         timeoutInMinutes: 80
         pool:
           vmImage: 'macOS-10.14'
         variables:
-          OSX_VERSION: '10.15'
+          build_dir: '$(Build.StagingDirectory)/build'
+          install_dir: '$(Build.StagingDirectory)/install'
         steps:
           - checkout: self
             submodules: true
           - bash: |
+              set -ex
               echo "##vso[task.prependpath]$CONDA/bin"
-              echo "##vso[task.setvariable variable=conda_dir]$CONDA"
-            displayName: 'Configure Conda directories'
-          - bash: |
               sudo chown -R $USER $CONDA
-            displayName: 'Take ownership of Conda installation'
+            displayName: 'Add Conda to PATH'
           - bash: |
               set -ex
               conda --version
-              conda install --yes conda-build
-            displayName: 'Conda configuration'
+              conda env create -f scipp-developer-no-mantid.yml
+              # TODO
+              echo "##vso[task.prependpath]$CONDA/envs/scipp-developer/bin"
+              mkdir "$(build_dir)" "$(install_dir)"
+            displayName: 'Configure build environment'
           - bash: |
-              conda build -c conda-forge ./conda
-            displayName: 'Package build'
-          - task: PublishBuildArtifacts@1
-            inputs:
-              PathtoPublish: '$(conda_dir)/conda-bld/osx-64'
-              ArtifactName: 'osx-64'
-            displayName: 'Publish Conda package artefacts'
+              set -ex
+              cd "$(build_dir)"
+              cmake \
+                -G Ninja \
+                -DPYTHON_EXECUTABLE=$(command -v python) \
+                -DCMAKE_INSTALL_PREFIX="$(install_dir)" \
+                -DCMAKE_BUILD_TYPE=Release \
+                "$(Build.Repository.LocalPath)"
+# -DCMAKE_OSX_DEPLOYMENT_TARGET='10.15' \
+# -DCMAKE_OSX_SYSROOT='/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk' \
+            displayName: 'CMake configuration'
+          - bash: |
+              set -ex
+              cd "$(build_dir)"
+              cmake --build . --target all-tests
+              ./common/test/scipp-common-test
+              ./units/test/scipp-units-test
+              ./core/test/scipp-core-test
+              ./variable/test/scipp-variable-test
+              ./dataset/test/scipp-dataset-test
+              ./neutron/test/scipp-neutron-test
+            displayName: 'C++ unit tests'
+          - bash: |
+              set -ex
+              cd "$(build_dir)"
+              cmake --build . --target install
+            displayName: 'Build Python library'
+          - bash: |
+              set -ex
+              cd "$(build_dir)"
+              PYTHONPATH="$(install_dir)" python -m pytest
```

Comment:
I can reproduce the problem. I think the mistake here is that you `cd` to the build directory (assuming the default directory is the original source directory). The pybind11 source is copied into the ouput directory and contains tests, but the python tests under `scipp/python/tests` are not copied there. I think you either need to eliminate the call to `cd` above (as I think you do for the linux builds) OR provide `"$(Build.Repository.LocalPath)"` as the directory argument to `pytest`

Reply:
:man_facepalming: